app/Http/Controllers/Front/PageController.php

```diff
@@ -766,35 +787,42 @@ public function getPriceDescription(int $productid)
     {
         try {
             $product = Product::find($productid);
-            if ($product['add_to_contact'] == 1) {
+
+            if ($product && $product->add_to_contact == 1) {
                 return '';
             }
 
             $priceDescription = '';
 
-            $plans = Plan::where('product', $productid)->get();
+            // Retrieve plans with eager loading of planPrice relationship
+            $plans = Plan::with('planPrice')->where('product', $productid)->get();
 
-            if ($plans) {
-                foreach ($plans as $plan) {
-                    if ($plan->days == 365 || $plan->days == 366) {
-                        $description = $plan->planPrice->first();
-                        if ($description->price_description == 'Free') {
-                            $priceDescription = 'free';
+            foreach ($plans as $plan) {
+                if ($plan->days == 365 || $plan->days == 366) {
+                    // Check if planPrice relationship is loaded and retrieve the first planPrice
+                    $description = $plan->planPrice->first();
+
+                    if ($description && $description->price_description == 'Free') {
```

Comment:
i think this condition is not right

Reply:
Yes, the check if ($description->price_description === 'Free') { return 'free'; } is correct for free products,when the price_description of the plan is 'Free'.

Reply:
I think we should check price since that is solid, description can change

Reply:
Got it